src/main/scala/com/fulcrumgenomics/vcf/api/VcfHeader.scala

```diff
@@ -0,0 +1,175 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.vcf.api
+
+import com.fulcrumgenomics.FgBioDef._
+import enumeratum.EnumEntry
+import htsjdk.samtools.{SAMSequenceDictionary, SAMSequenceRecord}
+
+import scala.collection.immutable
+
+/** Trait used to represent how many instances of an attribute are to be expected in a VCFs INFO or genotype fields.*/
+sealed trait VcfCount { }
+
+object VcfCount {
+  /** VcfCount that indicates there should be exactly one value per alternate allele on the Variant. */
+  case object OnePerAltAllele  extends VcfCount
+  /** VcfCount that indicates there should be exactly one value per allele, including reference, on the Variant. */
+  case object OnePerAllele     extends VcfCount
+  /** VcfCount that indicates there should be exactly one value per possible genotype.  E.g. for a variant with a
+    * ref allele and one alt allele, and a diploid sample, there would be three values. */
+  case object OnePerGenotype   extends VcfCount
+  /** The attribute may have any number of values. */
+  case object Unknown          extends VcfCount
+  /** The attribute must have a fixed number of values, indicated by `count`. */
+  case class Fixed(count: Int) extends VcfCount
+}
+
+/** Trait for the set of acceptable atomic types that can appear in VCF's INFO and genotype fields. */
+sealed trait VcfFieldType extends EnumEntry {
+  /** Must parse a String into a single value of the type given. */
+  def parse(s: String): Any
+}
+
+object VcfFieldType extends FgBioEnum[VcfFieldType] {
+  /** Integer field type, represented internally as [[Int]]. */
+  case object Integer   extends VcfFieldType { override def parse(s: String): Any = if (s == ".") -1 else s.toInt }
```

Comment:
This is horrifying.  Apparently it's ok to have an `Int` type field in VCF that takes multiple values and put in things like `1,2,.,4` - which makes it very hard to present typed values for array types.
Perhaps the thing to do here is replace the use of `IndexedSeq` with our own `MultiValuedAttr` type that wraps an array and also tracks missing values?  Ugh.